### PR TITLE
chore(schema): Descriptions

### PR DESCRIPTION
**Pull request by oded-soffrin**
Another fun task for this week is to document our whole schema to unblock blog
I started to process feel free to review, I'll make a commit with every file I'm done with

### CODE REVIEW

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto [hunk 1]
```diff
@@ -6,71 +6,71 @@ import "google/protobuf/wrappers.proto";
...
+        LEFT = 1; // Left Align
+        RIGHT = 2; // Right Align
+        CENTER = 3; // Center Align
+        JUSTIFY = 4; // Stretches the lines so that each line has equal width (like in newspapers and magazines)
     }
 
-    TextAlignment text_alignment = 1;
-    google.protobuf.StringValue line_height = 2; // CSS value
+    TextAlignment text_alignment = 1; // Text Alignment
+    google.protobuf.StringValue line_height = 2; // CSS LineHeight value
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto:17]: unitless
```suggestion
    google.protobuf.StringValue line_height = 2; // CSS LineHeight value (unitless ratio)
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto [hunk 2]
```diff
@@ -6,71 +6,71 @@ import "google/protobuf/wrappers.proto";
 
 message TextStyle {
     enum TextAlignment {
-        AUTO = 0;
-        LEFT = 1;
-        RIGHT = 2;
-        CENTER = 3;
-        JUSTIFY = 4;
+        AUTO = 0; // Alignment will depend on the language
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto:9]:
```suggestion
        AUTO = 0; // browser default, eqivalent to `initial`
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto [hunk 3]
```diff
@@ -6,71 +6,71 @@ import "google/protobuf/wrappers.proto";
...
+        LEFT = 1; // Left Align
+        RIGHT = 2; // Right Align
+        CENTER = 3; // Center Align
+        JUSTIFY = 4; // Stretches the lines so that each line has equal width (like in newspapers and magazines)
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto:13]:
```suggestion
        JUSTIFY = 4; // Text is spaced to line up its left and right edges to the left and right edges of the line box, except for the last line
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto [hunk 4]
```diff
@@ -79,30 +79,31 @@ message FileSource {
...
 }
 
 message Media {
     FileSource src = 1;
-    google.protobuf.Int32Value width = 2;
-    google.protobuf.Int32Value height = 3;
-    google.protobuf.DoubleValue duration = 4;
+    google.protobuf.Int32Value width = 2; // Media width in pixels 
+    google.protobuf.Int32Value height = 3; // Media height in pixels
+    google.protobuf.DoubleValue duration = 4; // Duration in seconds (when applied)
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto:89]:
```suggestion
    google.protobuf.DoubleValue duration = 4; // Duration in seconds (relevant to audio/video)
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/decoration_color.proto [hunk 5]
```diff
@@ -5,6 +5,6 @@ package wix.rich_content.v1;
 import "google/protobuf/wrappers.proto";
 
 message ColorData {
-    google.protobuf.StringValue background = 1;
-    google.protobuf.StringValue foreground = 2;
+    google.protobuf.StringValue background = 1; // CSS color value
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/decoration_color.proto:8]:
```suggestion
    google.protobuf.StringValue background = 1; // hexadecimal background color
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/decoration_color.proto [hunk 6]
```diff
@@ -5,6 +5,6 @@ package wix.rich_content.v1;
...
+    google.protobuf.StringValue foreground = 2; // CSS color value
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/decoration_color.proto:9]:
```suggestion
    google.protobuf.StringValue foreground = 2; // hexadecimal foreound color
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_button.proto [hunk 7]
```diff
@@ -8,28 +8,28 @@ import "google/protobuf/wrappers.proto";
 message ButtonData {
     message Styles {
         message Colors {
-            google.protobuf.StringValue text = 1; // hex color
-            google.protobuf.StringValue border = 2; // hex color
-            google.protobuf.StringValue background = 3; // hex color
+            google.protobuf.StringValue text = 1; // text hex color
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_button.proto:11]:
```suggestion
            google.protobuf.StringValue text = 1; // text hexadecimal color
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_button.proto [hunk 8]
```diff
@@ -8,28 +8,28 @@ import "google/protobuf/wrappers.proto";
...
+            google.protobuf.StringValue border = 2; // border hex color
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_button.proto:12]:
```suggestion
            google.protobuf.StringValue border = 2; // border hexadecimal color
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_button.proto [hunk 9]
```diff
@@ -8,28 +8,28 @@ import "google/protobuf/wrappers.proto";
...
+            google.protobuf.StringValue background = 3; // background hex color
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_button.proto:13]:
```suggestion
            google.protobuf.StringValue background = 3; // background hexadecimal color
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_button.proto [hunk 10]
```diff
@@ -8,28 +8,28 @@ import "google/protobuf/wrappers.proto";
...
-        Colors colors = 2;
+        Border border = 1; // Button's border attributes
+        Colors colors = 2; // Button's color attributes
     }
 
     enum Type {
-        LINK = 0;
-        ACTION = 1;
+        LINK = 0; // Regular link button
+        ACTION = 1; // Dynamic action button (need to be configured in consumption)
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_button.proto:26]:
```suggestion
        ACTION = 1; // Triggers custom action that is defined in plugin configuration by the consumer
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_collapsible_list.proto [hunk 11]
```diff
@@ -7,16 +7,16 @@ import "google/protobuf/wrappers.proto";
...
-        RTL = 1;
+        LTR = 0; // Left-to-right
+        RTL = 1; // Right-to-left
     }
-    PluginContainerData container_data = 1;
-    google.protobuf.BoolValue expand_only_one = 2;
-    InitialExpandedItems initial_expanded_items = 3;
-    Direction direction = 4;
+    PluginContainerData container_data = 1; // Collapsible List container data
+    google.protobuf.BoolValue expand_only_one = 2; // Collapse all items upon expending new item
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_collapsible_list.proto:19]:
```suggestion
    google.protobuf.BoolValue expand_only_one = 2; // Allows to expand single item simultaneously

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_gallery.proto [hunk 12]
```diff
@@ -9,81 +9,81 @@ import "google/protobuf/wrappers.proto";
...
+      ROWS = 0; // Rows Orientation
+      COLUMNS = 1; // Columns Orientation
     }
   
-    Type type = 1;
-    google.protobuf.BoolValue horizontal_scroll = 2;
-    Orientation orientation = 3;
-    google.protobuf.Int32Value number_of_columns = 4;
-    google.protobuf.Int32Value mobile_number_of_columns = 5;
+    Type type = 1; // Gallry type
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_gallery.proto:29]:
```suggestion
    Type type = 1; // Gallery layout type
```

Comment (lxgreen) [reply packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_gallery.proto:29]: it's the layout Type enum from line 11, right?

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_map.proto [hunk 13]
```diff
@@ -6,33 +6,27 @@ import "wix/rich_content/v1/common.proto";
...
-  // Visibilty of view mode control
-  google.protobuf.BoolValue viewModeControl = 9;
-  google.protobuf.Int32Value initialZoom = 10;
-  MapType mapType = 11;
+  google.protobuf.StringValue address = 1 [(.wix.api.pii) = true]; // exact address 
+  google.protobuf.BoolValue draggable = 2;  // Map Draggable option toggle
+  google.protobuf.BoolValue marker = 3; // Visibilty of marker  
+  google.protobuf.BoolValue streetViewControl = 4; // Visibilty of street view control  
+  google.protobuf.BoolValue zoomControl = 5; // Visibilty of zoom control  
+  google.protobuf.DoubleValue lat = 6; // latitude
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_map.proto:26]:
```suggestion
  google.protobuf.DoubleValue lat = 6; // Location latitude  
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_map.proto [hunk 14]
```diff
@@ -6,33 +6,27 @@ import "wix/rich_content/v1/common.proto";
...
-  google.protobuf.BoolValue viewModeControl = 9;
-  google.protobuf.Int32Value initialZoom = 10;
-  MapType mapType = 11;
+  google.protobuf.StringValue address = 1 [(.wix.api.pii) = true]; // exact address 
+  google.protobuf.BoolValue draggable = 2;  // Map Draggable option toggle
+  google.protobuf.BoolValue marker = 3; // Visibilty of marker  
+  google.protobuf.BoolValue streetViewControl = 4; // Visibilty of street view control  
+  google.protobuf.BoolValue zoomControl = 5; // Visibilty of zoom control  
+  google.protobuf.DoubleValue lat = 6; // latitude  
+  google.protobuf.DoubleValue lng = 7; // longitude
```
Comment (lxgreen) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_map.proto:27]:
```suggestion
  google.protobuf.DoubleValue lng = 7; // Location longitude
```

**Review by lxgreen — CHANGES_REQUESTED** [review]

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto [hunk 15]
```diff
@@ -79,30 +79,31 @@ message FileSource {
         google.protobuf.StringValue custom = 2 [deprecated = true]; // Custom ID
         google.protobuf.StringValue id = 3; // ID that will be resolved to a URL by a resolver function
     }
-    google.protobuf.BoolValue private = 4;
+    google.protobuf.BoolValue private = 4; // indicates whather the file source is public/private
```
Comment (maimonav) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto:82]: whether* :)

**Pull request merged**